pkg/sql/schemachanger/sctest/end_to_end.go

```diff
@@ -93,8 +94,12 @@ func EndToEndSideEffects(t *testing.T, dir string, newCluster NewClusterFunc) {
 				return sctestutils.Diff(a, b, sctestutils.DiffArgs{CompactLevel: 1})
 
 			case "test":
-				require.Len(t, stmts, 1)
-				stmt := stmts[0]
+				require.Lessf(t, numTestStatementsObserved, 1, "only one test per-file.")
+				numTestStatementsObserved++
+				stmtSqls := make([]string, 0, len(stmts))
+				for _, stmt := range stmts {
+					stmtSqls = append(stmtSqls, stmt.SQL)
```

Comment:
Can you help me understand this change, as it pertains to the `sctestdeps.WithStatements(stmt.SQL)` change below?

Reply:
This allows the test command to have multiple SQL statements. Below then we execute all the statements in a single transaction.

Reply:
In what case would we need this capability? I also just see one sql stmt under `test` in all the logic test file below. Did I miss something?

Reply:
For transaction testing, it allows us to have a test statement with multiple statements. Only one test is allowed per file.